File: java/TwoSum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] result = {0, 0};
        for (int i = 0; i < numbers.length; i++) {
            if (map.containsKey(numbers[i]) == false) {
                map.put(target-numbers[i], i);
            } else {
                result[0] = map.get(numbers[i]) + 1;
                result[1] = i + 1;
            }
        }
        return result;
    }
}

File: archive/cpp/TwoSum.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int> &numbers, int target) {
        std::vector<int> result;
        std::unordered_map<int, int> map;
        for (int i = 0; i < (int)numbers.size(); i ++) {
            if (map.find(numbers[i]) != map.end()) {
                result.push_back(map.find(numbers[i])->second + 1);
                result.push_back(i+1);
            } else {
                map[target-numbers[i]] = i;
            }
        }
        return result;
    }
};
